rfd/0163-vnet.md

```diff
@@ -904,6 +895,40 @@ message UpdateVNetRequest {
 message DeleteVNetRequest {}
 ```
 
+#### Daemon service
+
+```proto
+service VNetRootDaemon {
+  // Start creates and configures a TUN device and configures DNS for all active profiles found in
+  // tsh_home. It also saves pid and uid of the client. Until the client calls Stop or exits, no
+  // other client can call Start or Stop.
+  //
+  // Start returns the fd of the TUN device so that the client can read and write from it. This is
+  // done through the underlying socket connection, see https://pkg.go.dev/github.com/edwarnicke/grpcfd.
+  rpc Start(StartRequest) returns (StartResponse);
+
+  // RefreshDNSConfig re-fetches cluster information and updates the existing DNS configuration to
+  // account for any changes. Typically called by the client after it detects that the user has
+  // logged in or out of a cluster.
+  rpc RefreshDNSConfig(RefreshDNSConfigRequest) returns (RefreshDNSConfigResponse);
+
+  // Stop closes any active connections, closes the TUN device and cleans up DNS configuration.
+  // This is also done automatically if the daemon detects that the client has exited without
+  // explicitly calling Stop.
+  rpc Stop(StopRequest) returns (StopResponse);
+}
```

Comment:
I think the root daemon should watch TELEPORT_HOME and react to changes, updating DNS configuration as necessary when new profiles log in or expire. It will do this for `tsh_vnet`. If we do it that way we basically don't need a gRPC service for the daemon - it's either running or not running. But I think it's enough of an implementation detail we don't need it to be exactly right in the RFD, we can figure it out in a PR and update here